src/search_algo.hpp

```diff
@@ -443,10 +481,87 @@ seedLooksPromising(LocalDataHolder<TGlobalHolder> const & lH,
+// WARNING the following function only works for hammingdistanced seeds
+template <typename TGlobalHolder>
+inline bool
+seedLooksPromisingRev(LocalDataHolder<TGlobalHolder> const & lH,
+                      typename TGlobalHolder::TMatch const & m)
+{
+    int64_t effectiveQBegin = m.qryStart;
+    int64_t effectiveSBegin = m.subjStart;
+    uint64_t actualLength = m.qryEnd - m.qryStart;
+    uint64_t effectiveLength = std::max(static_cast<uint64_t>(lH.options.seedLength * lH.options.preScoring),
+                                        actualLength);
+
+    if (effectiveLength > actualLength)
+    {
+        effectiveQBegin -= (effectiveLength - actualLength) / 2;
+        effectiveSBegin -= (effectiveLength - actualLength) / 2;
+
+        int64_t min = std::min(effectiveQBegin, effectiveSBegin);
+        if (min < 0)
+        {
+            effectiveQBegin -= min;
+            effectiveSBegin -= min;
+            effectiveLength += min;
+        }
+
+        effectiveLength = std::min({
+                            static_cast<uint64_t>(std::ranges::size(lH.transQrySeqs[m.qryId]) - effectiveQBegin),
+                            static_cast<uint64_t>(std::ranges::size(lH.gH.transSbjSeqs[m.subjId]) - effectiveSBegin),
+                            effectiveLength});
+    }
+
+    auto const & qSeq = lH.transQrySeqs[m.qryId]
+                      | seqan3::views::slice(effectiveQBegin, static_cast<int64_t>(effectiveQBegin + effectiveLength));
+    auto const & sSeq = lH.gH.transSbjSeqs[m.subjId]
+                      | seqan3::views::slice(effectiveSBegin, static_cast<int64_t>(effectiveSBegin + effectiveLength));
+
+    int             s = 0;
+    int      maxScore = 0;
+    int const thresh  = lH.options.preScoringThresh * effectiveLength;
+
+    // score the diagonal
+    for (uint64_t i = 0; i < effectiveLength; ++i)
+    {
+        s += lH.gH.scoringSchemePreScoringRev.score(qSeq[i], sSeq[i]);
+
+        if (s < 0)
+            s = 0;
+        else if (s > maxScore)
+            maxScore = s;
+
+        if (maxScore >= thresh)
+            return true;
+    }
+
+    return false;
+}
+
+/*** THIS ACCESSES seqan3::detail:: IT SHOULD BE REPLACED WITH OFFICIAL INTERFACE ONCE OPTIMISED ***/
+template <typename TGlobalHolder, typename TSeed>
+inline void
+search_impl_fwd(LocalDataHolder<TGlobalHolder> & lH, TSeed && seed)
+{
+    seqan3::detail::search_param max_error{static_cast<uint8_t>(lH.options.maxSeedDist), // total
+                                           static_cast<uint8_t>(lH.options.maxSeedDist), // substitutions
+                                           0,                                            // insertions
+                                           0};                                           // deletions
+
+    auto delegate = [&lH] (auto const & it)
+    {
+        lH.cursor_buffer_fwd.push_back(it);
+    };
+
+    seqan3::detail::search_algo<false>(lH.gH.indexFileFwd.index, seed, max_error, delegate);
+}
+
```

Comment:
I would really like to prevent this duplication of logic here. If the same type is used for the scoring scheme (just two different objects), one could simply use one `seedLooksPromising()` function and add this at the beginning:
```cpp
auto const & scoringScheme = runtime_variable ? lH.gH.scoringSchemePreScoringRev :  lH.gH.scoringSchemePreScoringFwd;
```
And then use that later on.

Reply:
Especially since this function works with transSeqs and not redSeqs